### DISCUSSION TITLE
Dynamically group players

### DISCUSSION THREAD

**Discussion by wjbeckett** [music-assistant/support#410] [Feature requests and ideas]
I'd love to be able to dynamically create speaker groups for use in MA.
having a checkbox or something next to the output list to select multiple speakers to send the music to.
Could this be some sort of dynamically created helper group?

**Comment by marcelveldt** [comment]
I like the idea and I guess you'd like to control this from Home Assistant service calls (e.g. automations) ?
From the Music Assistant interface this is already possible:

1) In Home Assistant go to integrations - helpers
2) Create a group helper, type media player
3) Add the players you want
4) Add this new groupplayer to Music Assistant

The default behavior of MA is that it will only start streaming to those players that are powered on in that group.
So you can just power on some of the players in that group and start music on the group player. Voila.

**Reply by richardwonka** [reply]
I'd like to have the option of joining / unjoining speakers to speaker groups in the interface. 

When I look at the volume sliders:
- In groups: A minus **(-)** button next to each speaker in a group to call `media_player.unjoin` 
- for individual players: A **(+)** button to call `media_player.join` to the currently playing speaker or group would do the trick.

**Comment by johanneswilkens** [comment]
It would be nice if this could also be done by using the media_player.join and media_player.unjoin services, I think by using the helper it will start playing music on the tv when someone is watching something, and you want to play music in 2 rooms (which are in a group with the tv)?

**Comment by erkr** [comment]
I move this request to the discussions, thanks for filing.

Related to feature #385

**Comment by AxelJust** [comment]
Hey,
I setup multiple Squeezelite-ESP32 Boards in a Helper group, but they do not Playback synced. The test was done with Spotify.
with a LMS server it was flawles.

**Reply by RosensRauk** [reply]
So a hardware issue?
I would love if HA could sync to several speakers but my delay over DLNA over WIFI is not a constant. Regarding WIFI,  if using a couple of microfons for measuring the delta?

**Comment by JimmyGiant** [comment]
First of all, thanks for all the work. MASS is already providing so much functionality which I've long been looking for in HA.

> The default behavior of MA is that it will only start streaming to those players that are powered on in that group.
So you can just power on some of the players in that group and start music on the group player. Voila.

@marcelveldt: 
That actually works fine. But I can't get the group to unjoin afterwards. I disabled all speakers of the group individually, then the group itself.. When I start another player (outside of the group), the group turns back on again. Any thoughts on how to work around that?

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
